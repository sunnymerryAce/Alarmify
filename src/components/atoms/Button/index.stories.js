import React from 'react';
import '../../../App.css';
import Button from './index';

export default {
  title: 'Button',
};

export const Default = () => (
  <Button onClick={() => {}} text="Insert  Some Text" />
);
