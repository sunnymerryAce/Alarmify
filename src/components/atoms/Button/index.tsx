import React from 'react';
import styled from 'styled-components';

interface Props {
  onClick: (props: any) => void;
  text: string;
}

const Button: React.FC<Props> = (props) => {
  return (
    <StyledButton className="Button" onClick={props.onClick}>
      {props.text}
    </StyledButton>
  );
};

export default Button;

const StyledButton = styled.div`
  background-color: #1db954;
  font-size: 20px;
  font-weight: bold;
  line-height: 30px;
  border-radius: 50px;
  width: 70%;
  padding: 10px 0;
  margin: 7vh auto;
`;
