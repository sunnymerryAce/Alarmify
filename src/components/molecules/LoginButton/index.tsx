import React from 'react';
import Button from '../../atoms/Button';
import CONFIG from '../../../util/CONFIG';
import createURLSearchParamsFromObject from '../../../util/functions/createURLSearchParamsFromObject';

const LoginButton: React.FC = () => {
  const redirectURL = (): string => {
    const url = new URL(CONFIG.SPOTIFY_LOGIN_URI);
    const params = createURLSearchParamsFromObject({
      response_type: 'code',
      client_id: CONFIG.SPOTIFY_CLIENT_ID,
      redirect_uri: CONFIG.REDIRECT_URI,
      scope:
        'user-read-private user-read-playback-state user-modify-playback-state playlist-read-collaborative',
    });
    url.search = params.toString();
    return url.href;
  };

  const login = () => {
    window.location.href = redirectURL();
  };

  return (
    <div className="LoginButton">
      <Button onClick={login} text="LOGIN TO SPOTIFY"></Button>
    </div>
  );
};

export default LoginButton;
