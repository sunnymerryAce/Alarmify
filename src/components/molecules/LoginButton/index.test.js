import React from 'react';
import LoginButton, { redirectURL, login } from './index';
import Button from '../../atoms/Button';
import { shallow, mount } from 'enzyme';

const expectedUrl =
  'https://accounts.spotify.com/authorize?response_type=code&client_id=8be6bb9bbc644e93ade9e6ba983fa7b2&redirect_uri=https%3A%2F%2Falarmify-5f826.web.app%2F&scope=user-read-private+user-read-playback-state+user-modify-playback-state+playlist-read-collaborative';

describe('redirectURL', () => {
  test('should return proper url', () => {
    expect(redirectURL()).toBe(expectedUrl);
  });
});

describe('login', () => {
  test('proper url should be assigned to href', () => {
    global.window = Object.create(window);
    Object.defineProperty(window, 'location', {
      value: {
        href: '',
      },
    });
    login();
    expect(window.location.href).toBe(expectedUrl);
  });
});

describe('<LoginButton />', () => {
  const container = shallow(<LoginButton />);

  test('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot();
  });

  test('should have Button component', () => {
    expect(container.find(Button).length).toBe(1);
  });

  test('Button component should have proper props', () => {
    expect(container.find(Button).props()).toMatchObject({
      onClick: login,
      text: 'LOGIN TO SPOTIFY',
    });
  });
});

// 単体テストとしては不要？
describe('<LoginButton /> mount', () => {
  const containerMounted = mount(<LoginButton />);
  test('should have proper text', () => {
    expect(
      containerMounted
        .find('.Button')
        .hostNodes()
        .text(),
    ).toBe('LOGIN TO SPOTIFY');
  });
});
