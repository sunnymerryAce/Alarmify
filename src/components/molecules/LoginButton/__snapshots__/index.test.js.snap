// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginButton /> should match the snapshot 1`] = `"<div class=\\"LoginButton\\"><div class=\\"sc-bdVaJa lkGCXV Button\\">LOGIN TO SPOTIFY</div></div>"`;
